### ISSUE TITLE
Playground: elementToAppend is ignored in createTypeScriptSandbox

### ISSUE THREAD

**Issue by electrovir** [microsoft/TypeScript-Website#2565]
There is no way to use the `elementToAppend` property in `SandBoxConfig` when using `createTypeScriptSandbox` because of this line:

https://github.com/microsoft/TypeScript-Website/blob/95d050bf720aa352dbe5e0f4f1331323aa752398/packages/sandbox/src/index.ts#L116

`defaultPlaygroundSettings()` _always_ returns an object wherein `domID` is defined as an empty string so `domID` is _always_ in `config`.

**Repro:**

Call `createTypeScriptSandbox` with a config that attempts to utilize `elementToAppend` instead of `domID`.

**Expected behavior:**

I can use the `elementToAppend` property to create a sandbox.

**Actual behavior:**

`elementToAppend` is ignored.

**Comment by orta** [comment]
Yeah, this sounds like a bug, open to fixes - you can send in `domID: undefined` to work around it I assume?

**Comment by electrovir** [comment]
> you can send in domID: undefined to work around it I assume?

Unfortunately this does not work. The sandbox code does not check if `domID` is truthy, it checks if it exists in the config object:

https://github.com/microsoft/TypeScript-Website/blob/95d050bf720aa352dbe5e0f4f1331323aa752398/packages/sandbox/src/index.ts#L117

So even with `domID: undefined`, `elementToAppend` is still ignored. This then cascades into later errors as the sandbox tries to act with an undefined `domID`.

**Comment by jammawtf** [comment]
as a workaround you can decorate getElementById to return the element you want to append to : 
```
const oldGEBI = document.getElementById.bind(document)
document.getElementById = (id: string) => {
  if (id === 'monaco-editor-embed') {
    return editorDiv;
  }
  return oldGEBI(id);
}

const initsandbox = sandboxFactory.createTypeScriptSandbox({domID: 'monaco-editor-embed'}, main, window.ts);
document.getElementById = oldGEBI;
```

If element is not in the dom when the sandbox is created, you will have to invoke `sandbox.editor.layout()` after attaching it.

**Comment by agilgur5** [comment]
I believe #2885 is the same as this, and it is fixed by #2886
